### PR TITLE
DHFPROD-7058:Map multiple entities from one xml source doc

### PR DESCRIPTION
**Pull request by srinathgit**
### Description
Map multiple entities from one xml source doc
Added a test that maps multiple entities from a source xml doc and also updated reference-entity-model project with mapping from an xml doc
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

File: examples/reference-entity-model/input/xmlCustomersWithRelatedEntities/customer1WithRelatedEntities.xml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+      <OrderID>3002</OrderID>
+      <DateAndTime>2019/05/24-07:45:22</DateAndTime>
+      <FirstName>John</FirstName>
+      <LastName>Lithgow</LastName>
+      <SendToID>403</SendToID>
+      <Products>
+        <Product>
+          <Quantity>1</Quantity>
+          <ProductId>70</ProductId>
+          <Name>Skechers GOrun Razor +</Name>
```
Comment (rjrudin) [inline examples/reference-entity-model/input/xmlCustomersWithRelatedEntities/customer1WithRelatedEntities.xml:36]: For manual testing, it'll be easier to just name these e.g. "Product 70" and "Product 90". It'll be good to test longer names to see how they show up in the UI, but that's a separate concern.

File: examples/reference-entity-model/input/xmlCustomersWithRelatedEntities/customer1WithRelatedEntities.xml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      <City>Mountain view</City>
+      <State>California</State>
+      <Postal>94040-6783</Postal>
+    </Billing>
+  </Address>
+  <Orders>
+    <Order>
+      <OrderID>3002</OrderID>
+      <DateAndTime>2019/05/24-07:45:22</DateAndTime>
+      <FirstName>John</FirstName>
```
Comment (rjrudin) [inline examples/reference-entity-model/input/xmlCustomersWithRelatedEntities/customer1WithRelatedEntities.xml:29]: Why does Order have FirstName and LastName?

File: examples/reference-entity-model/steps/mapping/mapCustomersWithRelatedEntitiesXML.step.json [hunk 3]
```diff
@@ -0,0 +1,177 @@
+
+{
```
Comment (rjrudin) [inline examples/reference-entity-model/steps/mapping/mapCustomersWithRelatedEntitiesXML.step.json:2]: To keep this as easy to understand as possible, let's remove all the unnecessary properties here - additionalCollections, batchSize, validateEntity, headers, interceptors, provenance, customHook. Doing so makes it easier to see the stuff that matters.

File: marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/builtins/steps/mapping/map-multiple-entities/test-data/content/customerInfo.xml [hunk 4]
```diff
@@ -0,0 +1,41 @@
+<Customer xmlns="customerInfo">
```
Comment (rjrudin) [inline marklogic-data-hub/src/test/ml-modules/root/test/suites/data-hub/5/builtins/steps/mapping/map-multiple-entities/test-data/content/customerInfo.xml:1]: Same comments above about the test data. Also, let's start adding new tests under /data-hub/5/mapping - e.g. data-hub/5/mapping/map-multiple-entities . The builtins/steps directory should contain a minimum of code that delegates to libraries under data-hub/5/mapping - likewise for the data-services directory. 

The mapping app code is fine for now, we can move that later. But let's start getting the tests under data-hub/5/mapping.

**Review by rjrudin — CHANGES_REQUESTED** [review]

File: examples/reference-entity-model/input/xmlCustomersWithRelatedEntities/customer1WithRelatedEntities.xml (hunk 2 above)
Comment (srinathgit) [reply examples/reference-entity-model/input/xmlCustomersWithRelatedEntities/customer1WithRelatedEntities.xml:29]: I just based it on the example document in https://wiki.marklogic.com/display/PM/Mapping+Example+%28Data+Hub+5.5%29%3A+Map+multiple+entities+from+a+single+source. It's not used and I can remove them.

Comment (rjrudin) [reply examples/reference-entity-model/input/xmlCustomersWithRelatedEntities/customer1WithRelatedEntities.xml:29]: Yeah we want the example data and the mappings to be as lean and mean as possible - it makes the example as easy as possible to understand removes any noise from confusing folks.

**Review by rjrudin — APPROVED** [review]

**Review by ryanjdew — APPROVED** [review]

**Pull request merged**